fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/sfcc/flaps_channel.rs

```diff
@@ -252,7 +288,129 @@ impl FlapsChannel {
+                            self.conf1_flaps,
+                        ) =>
+                {
+                    self.flap_auto_command_angle = self.conf1_flaps
+                }
+                (_, Some(cas2))
+                    if cas2 > self.kts_100
+                        && cas2 < self.kts_210
+                        && SlatFlapControlComputerMisc::in_enlarged_target_range(
+                            self.flaps_feedback_angle,
+                            self.conf1_flaps,
+                        ) =>
+                {
+                    self.flap_auto_command_angle = self.conf1_flaps
+                }
+                // If one CAS <= 100 and other CAS >= 210 and flaps between 0 and 1+F, keep position
+                (Some(cas1), Some(cas2))
+                    if ((cas1 <= self.kts_100 && cas2 >= self.kts_210)
+                        || (cas1 >= self.kts_210 && cas2 <= self.kts_100))
+                        && SlatFlapControlComputerMisc::between_0_1f_enlarged_target_range(
+                            self.flaps_feedback_angle,
+                        ) =>
+                {
+                    // TODO: implement startup inhibition
+                    self.flap_auto_command_angle = self.flaps_feedback_angle
+                }
+                // If at least one CAS > 100 and < 210 and flaps between 0 and 1+F, keep position
+                (Some(cas1), _)
+                    if cas1 > self.kts_100
+                        && cas1 < self.kts_210
+                        && SlatFlapControlComputerMisc::between_0_1f_enlarged_target_range(
+                            self.flaps_feedback_angle,
+                        ) =>
+                {
+                    // TODO: implement startup inhibition
+                    self.flap_auto_command_angle = self.flaps_feedback_angle
+                }
+                (_, Some(cas2))
+                    if cas2 > self.kts_100
+                        && cas2 < self.kts_210
+                        && SlatFlapControlComputerMisc::between_0_1f_enlarged_target_range(
+                            self.flaps_feedback_angle,
+                        ) =>
+                {
+                    // TODO: implement startup inhibition
+                    self.flap_auto_command_angle = self.flaps_feedback_angle
+                }
+                (None, None) => self.flap_auto_command_angle = self.conf1f_flaps,
+                (_, _) => self.flap_auto_command_angle = self.flaps_feedback_angle,
+            }
+            self.flap_auto_command_active = true;
+        }
+    }
+
+    fn power_loss_reset(&mut self) {
+        self.flap_auto_command_active = false;
+        self.flap_auto_command_engaged = false;
+        self.flap_auto_command_angle = Angle::ZERO;
+    }
+
     fn generate_flap_angle(&mut self, adirs: &impl AdirsMeasurementOutputs) -> Angle {
+        if !self.is_powered_delayed.output() {
+            self.power_loss_reset();
```

Comment:
The reset logic should be placed in the `update` method not at a "random" place. When the calls are switched around later it might affect the behavior unintentionally.